Week-7/Assignment-7.3/loop.js

```diff
@@ -0,0 +1,46 @@
+
+
+let head;
+
+class Node {
+  constructor(val) {
+    this.data = val;
+    this.next = null;
+    this.flag = 0;
+  }
+}
+
+function push(data) {
+  let temp = new Node(data);
+  temp.next = head;
+  head = temp;
+}
+
+function loopCheck() {
+  let current = head;
+  while (current != null) {
+    if (current.flag == 1) {
+      return true;
+    }
+    current.flag = 1;
+    current = current.next;
+  }
+}
+
+push(10);
+push(20);
+push(30);
+push(40);
+push(50);
+
+//creating a loop here
```

Comment:
7.3 how will handle the case when head.next.next.next.next  fails as there can null referencing

Reply:
I have made some changes for 7.5 and 7.6 please verify once